src/SourceBuild/Arcade/eng/common/templates/job/source-build-build-tarball.yml

```diff
@@ -135,7 +135,11 @@ jobs:
         customBuildArgs="$customBuildArgs --poison"
       fi
 
-      docker run --rm -v $(tarballDir):/tarball -w /tarball ${networkArgs} $(_Container) ./build.sh --clean-while-building ${customBuildArgs} $(additionalBuildArgs)
+      if [[ '${{ parameters.usePreviousArtifacts }}' == 'true' || '$(_WithPreviousSDK)' == 'true' ]]; then
+        customBuildArgs="$customBuildArgs --with-sdk /tarball/.dotnet"
```

Comment:
I filed https://github.com/dotnet/source-build/issues/3190 to make this unnecessary.